src/fbrs/model/DatabaseModel.java

```diff
@@ -0,0 +1,112 @@
+package fbrs.model;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DatabaseModel {
+    private static DatabaseModel instance;
+    private DatabaseManager databaseManager = DatabaseManager.getInstance();
+    private List<Market> markets;
+    private List<Seller> sellers;
+    private List<Fisherman> fishermen;
+
+    private DatabaseModel() {
+    }
+
+    public static DatabaseModel getModel() {
+        if (instance == null)
+            instance = new DatabaseModel();
+        DatabaseManager.getInstance();
+        return instance;
+    }
+
+    public List<Market> getMarkets() {
+        if(markets == null) {
+            markets = DatabaseManager.getInstance().getMarkets();
+        }
+        return markets;
+    }
+
+    public String getMarkets(int marketId) {
+        if(markets == null) {
+            markets = DatabaseManager.getInstance().getMarkets();
+        }
+        for (Market market: markets) {
+            if (market.market_id == marketId)
+                return market.market_name;
+        }
+        return "خطأ في رقم السوق";
+    }
+
+    public int getMarkets(String marketName) {
+        if(markets == null) {
+            markets = DatabaseManager.getInstance().getMarkets();
+        }
+        for (Market market: markets) {
+            if (market.market_name.equals(marketName))
+                return market.market_id;
+        }
+        return -1;
+    }
```

Comment:
Why do you use linear scan to get a market?!?

Reply:
Names have been changed to reduce confuse
getMarkets(int marketId) ->getMarketByID(int marketId)
getMarkets(String marketName)->getMarketByName(String marketName)